apps/chat/src/components/Marketplace/TopicsList.tsx

```diff
@@ -1,175 +1,77 @@
-    const checkOverflow = () => {
-      if (containerRef.current && allTopicsRef.current) {
-        if (
-          allTopicsRef.current.getBoundingClientRect().width <=
-          containerRef.current.getBoundingClientRect().width
-        ) {
-          setVisibleTopics(topics);
-          setHiddenTopics([]);
-        } else {
-          const initialVisibleTopics: string[] = [];
-          const initialHiddenTopics: string[] = [];
-          const children = Array.from(allTopicsRef.current.children);
-          const containerWidth =
-            containerRef.current.getBoundingClientRect().width - extraSpace;
-          let occupiedWidth = 0;
-
-          const hiddenTopicWidths: { topic: string; topicWidth: number }[] = [];
-
-          children.forEach((childNode, index) => {
-            const element = childNode as HTMLElement;
-
-            const elementWidth = element.getBoundingClientRect().width;
-
-            if (occupiedWidth + elementWidth + topicGap <= containerWidth) {
-              initialVisibleTopics.push(topics[index]);
-              occupiedWidth += elementWidth + topicGap;
-            } else {
-              initialHiddenTopics.push(topics[index]);
-              hiddenTopicWidths.push({
-                topic: topics[index],
-                topicWidth: elementWidth,
-              });
-            }
-          });
-
-          setVisibleTopics(initialVisibleTopics);
-          setHiddenTopics(initialHiddenTopics);
-
-          let maxRowWidth = 0,
-            currentRowWidth = -topicGap;
-          for (const { topicWidth } of hiddenTopicWidths) {
-            if (
-              currentRowWidth + topicWidth + topicGap >
-              innerMaxTooltipWidth
-            ) {
-              maxRowWidth = Math.max(currentRowWidth, maxRowWidth);
-              currentRowWidth = topicWidth;
-            } else {
-              currentRowWidth += topicWidth + topicGap;
-            }
-          }
-          maxRowWidth = Math.max(currentRowWidth, maxRowWidth);
-
-          setMaxTooltipWidth(maxRowWidth);
-        }
-      }
-    };
-    const resizeObserver = new ResizeObserver(checkOverflow);
-
-    if (containerRef.current) {
-      resizeObserver.observe(containerRef.current);
-    }
-
-    return () => {
-      resizeObserver.disconnect();
-    };
-  }, [extraSpace, topics]);
   const handleDelayShowTooltip = useCallback((show: boolean) => {
     clearTimeout(timeoutRef.current);
     timeoutRef.current = setTimeout(
       () => setOpenHiddenTopics(show),
-      displayDelay,
+      DISPLAY_DELAY,
     );
   }, []);
 
-  return (
-    <>
-      <AllTopics topics={topics} allTopicsRef={allTopicsRef} />
-      <div
-        className="flex w-full gap-2"
-        ref={containerRef}
-        data-qa="app-topics"
+  const renderOverflow = (hiddenTopics: string[]) => (
+    <Tooltip
+      tooltip={
+        <div
+          className="my-1 flex flex-wrap gap-2"
+          style={{ maxWidth: `${maxTooltipWidth}px` }}
```

Comment:
```suggestion
          className="my-1 flex flex-wrap gap-2 max-w-[198px]"
```

and remove maxTooltipWidth